presto-hive/src/test/java/com/facebook/presto/hive/TestS3SelectRecordCursor.java

```diff
@@ -0,0 +1,213 @@
+    }
+
+    @Test
+    public void shouldReturnOnlyQuantityColumnInTheDDl()
+    {
+        String ddlSerializationValue = "struct article { varchar address varchar company date date_pub int quantity}";
+        String expectedDDLSerialization = "struct article { int quantity}";
+        assertEquals(buildSplitSchema(ddlSerializationValue, ARTICLE_COLUMN, QUANTITY_COLUMN),
+                buildExpectedProperties(expectedDDLSerialization, ARTICLE_COLUMN, QUANTITY_COLUMN));
+    }
+
+    @Test
+    public void shouldReturnProperties()
+    {
+        String ddlSerializationValue = "struct article { varchar article varchar author date date_pub int quantity}";
+        String expectedDDLSerialization = "struct article { varchar article, varchar author, date date_pub, int quantity}";
+        assertEquals(buildSplitSchema(ddlSerializationValue, DEFAULT_TEST_COLUMNS),
+                buildExpectedProperties(expectedDDLSerialization, DEFAULT_TEST_COLUMNS));
+    }
+
+    @Test
+    public void shouldReturnPropertiesWithoutDoubleCommaInColumnsName()
+    {
+        String ddlSerializationValue = "struct article { varchar article, varchar author, date date_pub, int quantity }";
+        String expectedDDLSerialization = "struct article { varchar article, varchar author, date date_pub, int quantity}";
+        assertEquals(buildSplitSchema(ddlSerializationValue, DEFAULT_TEST_COLUMNS),
+                buildExpectedProperties(expectedDDLSerialization, DEFAULT_TEST_COLUMNS));
+    }
+
+    @Test
+    public void shouldOnlyGetColumnTypeFromHiveObjectAndNotFromDDLSerial()
+    {
+        String ddlSerializationValue = "struct article { int article, double author, xxxx date_pub, int quantity }";
+        String expectedDDLSerialization = "struct article { int article, double author, xxxx date_pub, int quantity}";
+        assertEquals(buildSplitSchema(ddlSerializationValue, DEFAULT_TEST_COLUMNS),
+                buildExpectedProperties(expectedDDLSerialization, DEFAULT_TEST_COLUMNS));
+    }
+
+    @Test(expectedExceptions = NullPointerException.class)
+    public void shouldThrowNullPointerExceptionWhenColumnsIsNull()
+    {
+        updateSplitSchema(new Properties(), null);
+    }
+
+    @Test(expectedExceptions = NullPointerException.class)
+    public void shouldThrowNullPointerExceptionWhenSchemaIsNull()
+    {
+        updateSplitSchema(null, EMPTY_LIST);
+    }
+
+    private Properties buildSplitSchema(String ddlSerializationValue, HiveColumnHandle... columns)
+    {
+        Properties properties = new Properties();
+        properties.put(SERIALIZATION_LIB, APACHE_HADOOP_HIVE);
+        properties.put(SERIALIZATION_DDL, ddlSerializationValue);
+        return updateSplitSchema(properties, asList(columns));
+    }
+
+    private Properties buildExpectedProperties(String expectedDDLSerialization, HiveColumnHandle... expectedColumns)
+    {
+        final String expectedColumnsType = getTypes(expectedColumns);
```

Comment:
In general we don't mark local variables as final in Presto source code.

Reply:
Refactored